freud/diffraction.pyx

```diff
@@ -0,0 +1,316 @@
+# Copyright (c) 2010-2020 The Regents of the University of Michigan
+# This file is from the freud project, released under the BSD 3-Clause License.
+
+R"""
+The :class:`freud.diffraction` module provides functions for computing the
+diffraction pattern of particles in periodic systems.
+
+.. rubric:: Stability
+
+:mod:`freud.diffraction` is **unstable**. When upgrading from version 2.x to
+2.y (y > x), existing freud scripts may need to be updated. The API will be
+finalized in a future release.
+"""
+
+import freud.locality
+import logging
+import numpy as np
+import scipy.interpolate
+import scipy.ndimage
+import rowan
+import time
+
+from freud.util cimport _Compute
+cimport freud.util
+cimport numpy as np
+
+
+logger = logging.getLogger(__name__)
+
+
+cdef class DiffractionPattern(_Compute):
+    R"""Computes a 2D diffraction pattern.
+
+    The diffraction image represents the scattering of incident radiation,
+    and is useful for identifying translational order present in the system.
```

Comment:
```suggestion
    and is useful for identifying translational and/or orientational order present in the system.
```
See comment about quasicrystals. Also, I believe you can use diffraction to detect orientational order in nematic crystals (lacks translational order but has orienational order).

I'm not sure if 'and' or 'or' is more appropriate here.

Reply:
I'm not sure how we could detect orientational order when we don't use particles' orientations. I searched some literature to try and find diffraction images of nematic crystals. Some of the features that appear in those diffraction images _might_ come from the elliptical shape of the particles? I'm not 100% sure. Since this method treats all particles as point particles modeled by a Gaussian distribution, I don't want to make claims about detecting orientational/nematic order.

Reply:
When I added "orientational order", my intention was to include bond orientation than particle orientation. Maybe nematic crystals was a poor example and "rotational symmetry" would be more precise.

Reply:
Without incorporating a form factor I don't think you can say anything about orientations. If you do incorporate a form factor, then I think you may be able to see signal due to the correlations in nematic crystals, but I don't think this code should be attempting to support that right now.